azure/services/managedclusters/managedclusters.go

```diff
@@ -222,6 +260,28 @@ func (s *Service) Reconcile(ctx context.Context, spec interface{}) error {
 			KubernetesVersion: existingMC.ManagedClusterProperties.KubernetesVersion,
 		}
 
+		if managedCluster.AadProfile != nil {
+			propertiesNormalized.AadProfile = &containerservice.ManagedClusterAADProfile{
+				Managed:             managedCluster.AadProfile.Managed,
+				EnableAzureRBAC:     managedCluster.AadProfile.EnableAzureRBAC,
+				AdminGroupObjectIDs: managedCluster.AadProfile.AdminGroupObjectIDs,
+				ClientAppID:         managedCluster.AadProfile.ClientAppID,
+				ServerAppID:         managedCluster.AadProfile.ServerAppID,
+				ServerAppSecret:     managedCluster.AadProfile.ServerAppSecret,
+			}
+		}
+
+		if existingMC.AadProfile != nil {
```

Comment:
is there any normalization happening here? if no, you can simply compare the raw structs.

Reply:
ah, is tenant ID immutable?

Reply:
yeah amcp spec is normalized to aadProfile and version.